src/dag/chunk.rs

```diff
@@ -1,8 +1,100 @@
-pub type Key = String;
+use super::meta_generated;
+use flatbuffers;
+use flatbuffers::FlatBufferBuilder;
 
 #[allow(dead_code)]
+#[derive(Debug)]
 pub struct Chunk {
-    pub key: Key,
-    buf: Vec<u8>,
-    refs: [Key],
+    hash: String,
+    data: Vec<u8>,
+    meta: Option<(Vec<u8>, usize)>,
+}
+
+#[allow(dead_code)]
+impl Chunk {
+    pub fn new(hash: String, data: Vec<u8>, refs: &[&str]) -> Chunk {
+        Chunk {
+            hash,
+            data,
+            meta: Chunk::create_meta(refs),
+        }
+    }
+
+    pub fn read(hash: String, data: Vec<u8>, meta: Option<Vec<u8>>) -> Chunk {
+        Chunk {
+            hash,
+            data,
+            meta: meta.map(|v| (v, 0)),
+        }
+    }
+
+    pub fn hash(&self) -> &str {
+        &self.hash
+    }
+
+    pub fn data(&self) -> &[u8] {
+        &self.data
+    }
+
+    pub fn refs<'a>(&'a self) -> Box<dyn Iterator<Item = &str> + 'a> {
+        match self.meta() {
+            None => Box::new(std::iter::empty()),
+            Some(buf) => match meta_generated::meta::get_root_as_meta(buf).refs() {
+                None => Box::new(std::iter::empty()),
+                Some(v) => Box::new(v.iter()),
+            },
+        }
+    }
+
+    pub fn meta(&self) -> Option<&[u8]> {
+        match &self.meta {
+            None => None,
+            Some((buf, offset)) => Some(&buf[*offset..]),
+        }
+    }
+
+    fn create_meta(refs: &[&str]) -> Option<(Vec<u8>, usize)> {
+        if refs.is_empty() {
+            return None;
+        }
+        let mut builder = FlatBufferBuilder::default();
+        let refs = builder.create_vector_of_strings(refs);
```

Comment:
This allocates, but I'd already given up on the write path by that point.

Reply:
Should we start adding TODO comments for places like this where we suspect there's a better way and we just don't want to block on finding it right now? I think that may be appropriate.

Reply:
Done.